src/fluree/db/query/fql/syntax.cljc

```diff
@@ -0,0 +1,198 @@
+(ns fluree.db.query.fql.syntax
+  (:require [clojure.spec.alpha :as s]
+            [fluree.db.util.core :refer [pred-ident?]]))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(s/def ::limit pos-int?)
+
+(s/def ::offset nat-int?)
+
+(s/def ::maxFuel number?)
+(s/def ::max-fuel ::maxFuel)
+
+(s/def ::depth nat-int?)
+
+(s/def ::prettyPrint boolean?)
+(s/def ::pretty-print ::prettyPrint)
+
+(s/def ::parseJSON boolean?)
+(s/def ::parse-json ::parseJSON)
+
+(s/def ::opts (s/keys :opt-un [::maxFuel ::max-fuel ::parseJSON ::parse-json
```

Comment:
I love that this exists! How nice to see everything in one spot. One opt that is missing is the `::js?` key, which should be a boolean if it is present.